src/App.js

```diff
@@ -76,14 +67,16 @@ function App() {
   }, [isModalOpen, activeMentorId]);
 
   const [searchValue, setSearchValue] = useState("");
-  const searchChangeHandler = (input) => {
-    setSearchValue(input);
-    if (input.trim().length === 0) {
-      setVisibleMentorIds(mentorIds[waveIndex]);
-    }
-  };
-  const searchSelectHandler = (query) =>
-    setVisibleMentorIds(fieldSearch(query, waveIndex));
+  const [searchQuery, setSearchQuery] = useState({});
+  useEffect(() => setSearchValue(""), [waveIndex]);
+
+  const visibleMentorIds = useMemo(
+    () =>
+      searchValue.trim().length === 0
+        ? mentorIds[waveIndex]
+        : fieldSearch(searchQuery, waveIndex),
+    [searchQuery, searchValue, waveIndex]
+  );
```

Comment:
Wow this solution is pretty elegant. Guess I’m not too familiar with React’s hooks yet. 👍

Reply:
@bnjmnt4n Yeahh I learnt about the hooks during code review this week, which is why I thought of submitting this PR :joy: